src/apps/backend/modules/shared-task/internal/shared-task-util.ts

```diff
@@ -0,0 +1,55 @@
+import { SharedTask } from '../types';
+import { SharedTaskDB } from './store/shared-task-db';
+import { Task } from '../../task/types';
+import { Account } from '../../account/types';
+import { Types } from 'mongoose';
+
+export default class SharedTaskUtil {
+  public static convertSharedTaskDBToSharedTask(
+    sharedTaskDb: SharedTaskDB,
+  ): SharedTask {
+    const sharedTask = new SharedTask();
+    sharedTask.id = sharedTaskDb._id.toString();
+
+    if (Types.ObjectId.isValid(sharedTaskDb.task.toString())) {
+      sharedTask.task = sharedTaskDb.task.toString();
+    } else {
+      sharedTask.task = SharedTaskUtil.convertTask(sharedTaskDb.task);
```

Comment:
If you want `Task` after all, why are you not using `convertTaskDBToTask` from `TaskUtil`?

If you want something different from `Task` for this populated task then you can create a method like you have created `convertTask`.